### PR TITLE
fix(sharingDetails): Show correct share target

### PR DESCRIPTION
**Pull request by nfebe**
The share details page should show the target email or user/server to which
the user is creating the share.

| Before email | Before federation | After email | After federation |
| ----------------  | ---------------------- | --------------- | --------------------- |
| before-email | before-federation| after-email | after-federation |

### CODE REVIEW

**Comment by kesselb** [comment]
Please use the right type for your commit message. Refactoring is changing code without changing behavior.

**Review by artonge — APPROVED** [review]

File: apps/files_sharing/src/views/SharingDetailsTab.vue [hunk 1]
```diff
@@ -340,20 +340,25 @@ export default {
...
+					const [user, server] = this.share.shareWith.split('@')
+					return t('files_sharing', 'Share with {user} on remote server {server}', { user, server })
+				})()
 			case this.SHARE_TYPES.SHARE_TYPE_REMOTE_GROUP:
 				return t('files_sharing', 'Share with remote group')
 			case this.SHARE_TYPES.SHARE_TYPE_GUEST:
 				return t('files_sharing', 'Share with guest')
 			default:
-				return t('files_sharing', 'Share with')
+				return t('files_sharing', this.share.id ? 'Create share' : 'Update share')
```
Comment (susnux) [inline apps/files_sharing/src/views/SharingDetailsTab.vue:361]: This does not work as you might expect, this only works if you use exact the same strings in another place, as the l10n extraction does only handle plain strings.

```suggestion
				if (this.share.id) {
					// Share already exists
					return t('files_sharing', 'Update share')
				} else {
					return t('files_sharing', 'Create share')
				}
```

**Review by susnux — CHANGES_REQUESTED** [review]

File: apps/files_sharing/src/views/SharingDetailsTab.vue [hunk 2]
```diff
@@ -340,20 +340,32 @@ export default {
...
+			    return t('files_sharing', 'Share with email {email}', { email: this.share.shareWith })
 			case this.SHARE_TYPES.SHARE_TYPE_LINK:
 				return t('files_sharing', 'Share link')
 			case this.SHARE_TYPES.SHARE_TYPE_GROUP:
 				return t('files_sharing', 'Share with group')
 			case this.SHARE_TYPES.SHARE_TYPE_ROOM:
 				return t('files_sharing', 'Share in conversation')
 			case this.SHARE_TYPES.SHARE_TYPE_REMOTE:
-				return t('files_sharing', 'Share with remote')
+			    return (() => {
```
Comment (susnux) [inline apps/files_sharing/src/views/SharingDetailsTab.vue:352]: why is this a function?

Comment (Fenn-CS) [reply apps/files_sharing/src/views/SharingDetailsTab.vue:352]: Maintaining lexical structure of the switch case to avoid lint errors.

**Review by susnux — APPROVED** [review]
thank you!

**Pull request merged**